pkg/storage/kubernetes.go

```diff
@@ -61,7 +120,7 @@ func (k kubernetesClient) ListFromCluster() (StorageList, error) {
 			if volumeMount.Name == pvc.Name+"-vol" {
 				found = true
 				size := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
-				storage = append(storage, GetMachineFormatWithContainer(pvc.Labels[storagelabels.DevfileStorageLabel], size.String(), volumeMount.Spec.Path, volumeMount.Spec.ContainerName))
+				storage = append(storage, GetMachineFormatWithContainer(pvc.Labels[storagelabels.StorageLabel], size.String(), volumeMount.Spec.Path, volumeMount.Spec.ContainerName))
 			}
 		}
 		if !found {
```

Comment:
1. So during first time push, if there is a matching PVC with the labels, odo err out here? Interesting. I would think odo would clean up old(possibly stale PVC on first time push if they were left behind PVC with matching labels)

2. What if in the next section where you have a config with same vol spec, you may possibly reuse the vol (or probably not as it may not be in the same state 🤔). In this case, odo would just delete this volume and recreate it again.

Reply:
> So during first time push, if there is a matching PVC with the labels, odo errors out here? Interesting. I would think odo would clean up old(possibly stale PVC on first time push if they were left behind PVC with matching labels)

If there are PVCs with matching labels, we shouldn't clean the old ones as it might contain data which might be required by the user. So we error out and let the user know about it.

> What if in the next section, you have a config with same vol spec, you may possibly reuse the vol (or probably not as it may not be in the same state thinking). In this case, odo would just delete this volume and recreate it again.

We can't reuse as the user would expect a fresh volume without any previous data.